test/core/test/timeout.spec.ts

```diff
@@ -0,0 +1,10 @@
+import { describe, expect, test } from 'vitest'
+
+describe('suite timeout', () => {
+  test('true is true after 5100ms', async () => {
+    await new Promise(resolve => setTimeout(resolve, 5100))
+    expect(true).toBe(true)
+  })
+}, {
+  timeout: 6000,
+})
```

Comment:
This is also a valid syntax, which might be worth adding a test for (can't see that this case is handled in your code changes). However, it might be parsed in a different, but worth adding a test for it IMO!
```suggestion
describe('suite timeout simple input', () => {
  test('true is true after 5100ms', async () => {
    await new Promise(resolve => setTimeout(resolve, 5100))
    expect(true).toBe(true)
  })
}, 6000)
```

Reply:
Yes, we should also add a failing test case for this in `test/failing`

Reply:
Done.